bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/ui/QuickSearchDialog.java

```diff
@@ -417,8 +424,16 @@ protected void restoreDialog(IDialogSettings settings) {
 				pattern.setText(lastSearch);
 				pattern.selectAll();
 			}
-			if (settings.get(DIALOG_PATH_FILTER)!=null) {
-				String filter = settings.get(DIALOG_PATH_FILTER);
+
+			// Retrieve the last locations where the user searched (works across restarts)
+			filterHistory.addAll(List.of(settings.getArray(FILTER_HISTORY)));
+			if (!filterHistory.isEmpty()) {
+				String filter = filterHistory.getFirst();
+
+				// Filter out blanks
+				filterHistory.removeIf(String::isBlank);
+
+				searchIn.setItems(filterHistory.toArray(String[]::new));
```

Comment:
Nitpick: I don't think we need a comment to explain the next line
```suggestion
				String filter = filterHistory.getFirst();
				filterHistory.removeIf(String::isBlank);
				searchIn.setItems(filterHistory.toArray(String[]::new));
```

Reply:
I'd normally agree but seeing that the builds are so brittle, I'd like to leave it as it is.